sniffles/ruletrafficgenerator.py

```diff
@@ -985,6 +985,58 @@ def updateSequence(self, dir="to server", data_len=1):
                 self.current_seq_b_to_a += data_len
                 self.current_ack_a_to_b = self.current_seq_b_to_a
 
+# !!Create background traffic.
+class BackgroundTraffic(TrafficStream):
+    def __init__(self, rule=None, sconf=None, start_sec=-1, start_usec=0):
+        super().__init__()
+        self.proto = rule.getProto()
+        self.sport = Port(rule.getSport())
+        self.dport = rule.getDport()
+        self.rule = rule
+        self.rand = False
+        self.current_seq_a_to_b = random.randint(0, 4000000000)
+        self.current_ack_a_to_b = 0
+        self.current_seq_b_to_a = random.randint(0, 4000000000)
+        self.current_ack_b_to_a = 0
+
+        if sconf:
+            handshake = sconf.getTCPHandshake()
+            teardown = sconf.getTCPTeardown()
+            self.bi = sconf.getBi()
+            self.flow_ack = sconf.getTCPACK()
+            self.full_eval = sconf.getFullEval()
+            self.full_match = sconf.getFullMatch()
+            ipv6_percent = sconf.getIPV6Percent()
+            if sconf.getLatency() > 0:
+                self.latency = sconf.getLatency()
+            self.mac_def_file = sconf.getMacAddrDef()
+            if self.mac_def_file:
+                self.mac_gen = ETHERNET_HDR_GEN_DISTRIBUTION
+            self.pkt_len = sconf.getPktLength()
+            if sconf.getPktsPerStream() > 1:
+                self.packets_in_stream = sconf.getPktsPerStream()
+
+        if ipv6_percent > 0:
```

Comment:
As above, if you have passed in the sconf, I don't believe you need to worry about this either.  It will be handled in the super constuctor.